crates/annuity/src/lib.rs

```diff
@@ -129,11 +133,11 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
         T::BlockRewardProvider::distribute_block_reward(&annuity_pallet_id, reward_per_block)
     }
 
-    fn reward_per_block() -> BalanceOf<T, I> {
+    pub fn update_reward_per_block() {
         let emission_period = T::BlockNumberToBalance::convert(T::EmissionPeriod::get());
         let total_balance = T::Currency::total_balance(&Self::annuity_pallet_id());
         let reward_per_block = total_balance.checked_div(&emission_period).unwrap_or_default();
-        reward_per_block
+        RewardPerBlock::<T, I>::put(reward_per_block);
```

Comment:
We shouldn't cache this, the point of the annuity pallet is to use whatever free balance it has. This is actually less error-prone than your new approach since `distribute_block_reward` won't try to give a cached balance it doesn't have.

Reply:
Good catch @sander2 there is a bug in the calculation because we transfer, you're right in that this should be a linear function.